vpr/src/place/place.cpp

```diff
@@ -1518,6 +1535,45 @@ static e_move_result try_swap(float t,
 
     move_generator.process_outcome(move_outcome_stats);
 
+#ifdef VTR_ENABLE_DEBUG_LOGGING
+
+    t_draw_state* draw_state = get_draw_state_vars();
+    if (draw_state->list_of_breakpoints.size() != 0) {
+        //update current information
+        transform_blocks_affected(blocks_affected);
+        get_bp_state_globals()->get_glob_breakpoint_state()->move_num++;
+        get_bp_state_globals()->get_glob_breakpoint_state()->from_block = size_t(blocks_affected.moved_blocks[0].block_num);
+
+        //check for breakpoints
+        f_placer_debug = check_for_breakpoints(true);
```

Comment:
I don't think I understand how f_placer_debug is being used.  Is it something that a person needs to edit in the source code to turn on debugging, or a variable that's being used for communication inside the program?

Reply:
Basically, after every placement perturbation, the program checks if f_placer_debug is true, and if true, it stops the program until proceed is pressed again. Breakpoints stop the program by setting f_placer_debug to true.